Source/Core/Core/PowerPC/JitArm64/Jit.cpp

```diff
@@ -364,9 +365,26 @@ void JitArm64::WriteExit(u32 destination, bool LK, u32 exit_address_after_return
   if (LK)
   {
     // Push {ARM_PC+20; PPC_PC} on the stack
-    MOVI2R(ARM64Reg::X1, exit_address_after_return);
+    ARM64Reg reg_to_push;
+    // Push {ARM_PC, PPC_PC} on the stack
+    if (ARM64Reg::X1 != preloaded_after_reg)
+    {
+      if (preloaded_after_reg == ARM64Reg::INVALID_REG)
+      {
+        MOVI2R(ARM64Reg::X1, exit_address_after_return);
+        reg_to_push = ARM64Reg::X1;
+      }
+      else
+      {
+        reg_to_push = preloaded_after_reg;
+      }
+    }
+    else
+    {
+      reg_to_push = ARM64Reg::X1;
+    }
```

Comment:
Simpler to express as:

```suggestion
    if (preloaded_after_reg == ARM64Reg::INVALID_REG)
    {
      MOVI2R(ARM64Reg::X1, exit_address_after_return);
      reg_to_push = ARM64Reg::X1;
    }
    else
    {
      reg_to_push = preloaded_after_reg;
    }
```